src/resolvers.ts

```diff
@@ -0,0 +1,65 @@
+import 'reflect-metadata';
+import { AppDataSource } from './data-source';
+import { User } from './entity/User';
+import * as bcrypt from 'bcrypt';
+import { InputErrors } from './test/error';
+import Jwt from 'jsonwebtoken';
+import { CreateUserInput, LoginInput } from './schema';
+
+export const resolvers = {
+  Query: {
+    hello: () => 'Hello World',
+  },
+  Mutation: {
+    createUser: async (_: any, args: CreateUserInput) => {
+      //busca do email - busca no banco se já existe um email igual
+      const storageUsers = await AppDataSource.manager.findOne(User, { where: { email: args.input.email } });
+      if (storageUsers) {
+        throw new InputErrors('Email já registrado');
+      }
+      //validação da senha - 6 caracteres, com 1 letra e 1 digito
+      const validPassword = args.input.password;
+      const onlyCharacters = /\d/;
+      const onlyNumbers = /[a-zA-Z]/;
+      if (validPassword.length < 6 || !onlyCharacters.test(validPassword) || !onlyNumbers.test(validPassword)) {
+        throw new InputErrors('Senha inválida, deve conter ao menos 6 caracteres, 1 letra e um dígito');
+      }
+      //criação do hash da senha
+      const hashedPassword = await bcrypt.hash(args.input.password, 10);
+
+      //validação do email
+      const validateEmail = args.input.email;
+      const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+      if (!emailRegex.test(validateEmail)) {
+        throw new InputErrors('Email inválido');
+      }
+
+      const user = new User();
+      user.name = args.input.name;
+      user.email = args.input.email;
+      user.password = hashedPassword; //armazena o hash invés da senha
+      user.birthDate = args.input.birthDate;
+
+      return AppDataSource.manager.save(user);
+    },
+    login: async (_: any, args: LoginInput) => {
+      //bate no banco e acha um usuário pelo email
+      const user = await AppDataSource.manager.findOne(User, { where: { email: args.input.email } });
+      if (!user) {
+        throw new InputErrors('Usuário não encontrado');
+      }
+
+      //compara o hash do input e o hash do user que está no banco
+      const passwordMatch = await bcrypt.compare(args.input.password, user.password);
+      if (!passwordMatch) {
+        throw new InputErrors('Senha incorreta');
```

Comment:
Concorda que dizer para o usuario que a senha ta errada é uma porta para uma possivel falha de segurança? Isso porque dizer para um possivel invasor que a senha está errada, pode ser interpertado como "o email está certo, mas a senha está errada" (que no fim das contas é a verdade no contexto dessa funcao), facilitando algumas tecnicas de [força bruta](https://nordvpn.com/pt-br/blog/ataque-de-forca-bruta/#:~:text=O%20ataque%20de%20for%C3%A7a%20bruta,c%C3%B3digo%20de%20seguran%C3%A7a%20ou%20uma). Sempre importante lembrar da segurança no fazer um fluxo de login, e nesse sentido, significa generealizar até as mensagens de erro, por exemplo `"E-mail ou senha inválido"` ou `"Usuário sem credenciais válidas"`, mesmo que o erro não seja numa validacao de senha/email.